datatypes/src/collections/feature_collection_impl.rs

```diff
@@ -670,6 +709,44 @@ macro_rules! feature_collection_impl {
                 )
             }
         }
+
+        impl PartialEq for $Collection
+        where
+            Self: crate::collections::FeatureCollectionImplHelpers,
+        {
+            fn eq(&self, other: &Self) -> bool {
+                use crate::collections::feature_collection_impl::FeatureCollectionImplHelpers;
+                use crate::util::arrow::downcast_array;
+                use arrow::datatypes::DataType;
+
+                if self.types() != other.types() {
+                    return false;
+                }
+
+                for key in self.types().keys() {
+                    let c1 = self.table().column_by_name(key).expect("column must exist");
+                    let c2 = other
+                        .table()
+                        .column_by_name(key)
```

Comment:
is this correct? i guess two collections could have different column(names) ?
please add a test  where column names are not equal